ansible_base/lib/middleware/logging/log_request.py

```diff
@@ -1,10 +1,32 @@
 import logging
+import traceback
+import uuid
 
 from ansible_base.lib.logging import thread_local
 
 logger = logging.getLogger(__name__)
 
 
+class LogTracebackMiddleware:
+    transactions = {}
+
+    @classmethod
+    def handle_signal(cls, *args):
+        for t_id, request in LogTracebackMiddleware.transactions.items():
+            logger.error(f"Received graceful timeout signal for {request.method} path: {request.path} while in stack: {''.join(traceback.format_stack())}")
+
+    def __init__(self, get_response):
+        self.get_response = get_response
```

Comment:
(assuming we import signal above)
Are you suggesting this should work:

```suggestion
        self.get_response = get_response
        signal.signal(signal.SIGSYS, LogTracebackMiddleware.handle_signal)
```
?
Do you think I could even do
```suggestion
        self.get_response = get_response
        signal.signal(signal.SIGSYS, LogTracebackMiddleware.handle_signal)
        signal.signal(signal.SIGABRT, LogTracebackMiddleware.handle_signal)
```
and it would "just work"/handle either SIGSYS or SIGABRT ?

Reply:
I'm not convinced what you're doing is "safe" in the first place. What did this SIGSYS signal do _before_ you did this?

https://docs.python.org/3/library/signal.html#signal.signal

> The previous signal handler will be returned (see the description of [getsignal()](https://docs.python.org/3/library/signal.html#signal.getsignal) above). (See the Unix man page [signal(2)](https://manpages.debian.org/signal(2)) for further information.)

...suggesting we could just save the original handler and call that as a side effect of our method. But...

https://stackoverflow.com/questions/45181426/python-how-to-return-control-to-original-signal-handler-registered-from-c

Never mind, that's gross and terrible.

Maybe we could confirm that SIGSYS current signal is None if you do getsignal. If not, raise an exception. Then you could add a 1,000 different signals, as long as you're confirming in advance that you're not disconnecting an existing signal.

Reply:
so SIGSYS is just what I configured in the uwsgi conf for the graceful harakiri signal

Reply:
and SIGABRT is what gunicorn sends according to the **internet** https://ve3zsh.ca/blog/2021/05/15/graceful_gunicorn_timeouts.html